pkg/syncer/cnsoperator/controller/csinodetopology/csinodetopology_controller.go

```diff
@@ -365,3 +413,41 @@ func getNodeTopologyInfo(ctx context.Context, nodeVM *cnsvsphere.VirtualMachine,
 	}
 	return topologyLabels, nil
 }
+
+// findExistingTopologyLabels figures out if the given list of nodes were already participating
+// in a topology setup. If yes, it will discover if the nodes use the standard topology beta
+// labels or the GA labels and return those. However, if the cluster has a mix of nodes with beta
+// and GA labels, it will error out and request the customer to fix the environment before proceeding
+// further. If no topology labels were found on all nodes, it will return empty strings.
+func findExistingTopologyLabels(ctx context.Context, nodeList *corev1.NodeList) (string, string, error) {
+	log := logger.GetLogger(ctx)
+	labelMap := map[string]bool{
+		"beta": false,
+		"GA":   false,
+	}
+	for _, k8sNode := range nodeList.Items {
+		for key := range k8sNode.Labels {
+			if key == corev1.LabelTopologyZone || key == corev1.LabelTopologyRegion {
+				labelMap["GA"] = true
+			}
+			if key == corev1.LabelFailureDomainBetaZone || key == corev1.LabelFailureDomainBetaRegion {
+				labelMap["beta"] = true
+			}
+		}
+		if labelMap["GA"] && labelMap["beta"] {
+			return "", "", logger.LogNewErrorf(log, "found conflicting topology labels on certain node(s) in "+
+				"the cluster. Nodes in the cluster should either have the standard topology beta labels "+
+				"starting with \"failure-domain.beta.kubernetes.io\" or standard topology GA labels starting with "+
+				"\"topology.kubernetes.io\".")
+		}
```

Comment:
Logging nodes is not required. Just running `kubectl get nodes --show-labels` will tell the customer this info instead of browsing through logs. For debugging, we just need to know if there was a mix of labels or not.